src/Microsoft.DotNet.Core.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Core.Build.Tasks
+{
+    public class DependencyContextBuilder
+    {
+        public DependencyContextBuilder()
+        {
+        }
+
+        public DependencyContext Build(
+            string projectName,
+            string projectVersion,
+            CompilationOptions compilerOptions,
+            LockFile lockFile,
+            NuGetFramework framework,
+            string runtime)
+        {
+            LockFileTarget lockFileTarget = lockFile.GetTarget(framework, runtime);
+            IEnumerable<LockFileTargetLibrary> runtimeExports = lockFileTarget.Libraries;
+
+            // TODO: get this from the lock file once https://github.com/NuGet/Home/issues/2695 is fixed.
+            var packageName = "Microsoft.NETCore.App";
+            var platformExport = lockFileTarget
+                .Libraries
+                .FirstOrDefault(e => e.Name.Equals(packageName, StringComparison.OrdinalIgnoreCase));
+
+            bool portable = platformExport != null;
```

Comment:
There is no other way to tell from just a lock file if the app is portable or not.  Since the default, and first thing we need to get working is portable apps, this code is assuming the app is portable until we can get the info from the lock file.